src/xdsl/printer.py

```diff
@@ -115,15 +115,19 @@ def _print_message(self,
     T = TypeVar('T')
 
     def print_list(self,
-                   elems: FrozenList[T] | List[T],
+                   elems: Iterable[T],
                    print_fn: Callable[[T], None],
                    delimiter: str = ", ") -> None:
-        if len(elems) == 0:
+        it = iter(elems)
+        try:
+            el = next(it)
+            print_fn(el)
+            while True:
```

Comment:
You should be able to use a `for` loop here a well right?
After calling `el = next(it)`, `it` should be the iterator starting at the second element.

Reply:
I can't figure out to get a for loop to work, maybe I'm missing something. We only want the comma printed in between the element, so we need to either have a special case for the first or last element. This version prints the first element without a comma, then the next elements preceded by commas.

Reply:
I was thinking of something like:
```
it = iter(elems)
// Print the first element, if it exists
try:
  el = next(it)
except StopIteration:
  return
print_fn(el)
for el in it:
  self.print(delimiter)
  print_fn(el)
```